### PR TITLE
fix: send formdata on submit and add search url prop

### PR DESCRIPTION
**Pull request by BogdanDanilescu**
### HTML Header Component

- send the search query on the search functionality within the `Header` component.
- Add `searchUrl` prop in order to properly redirect the user to the search page

### React Header Component

- send the search query on the search functionality within the `Header` component.
- Add `searchUrl` prop in order to properly redirect the user to the search page
- Refactor and add props to make the header more customisable 
- Implement the ability to add extra buttons either with or without label/icon
- Add an optional title to the Header
Screenshots
image
image
image
image

### CODE REVIEW

File: packages/react/ds/src/header/header.tsx [hunk 1]
```diff
@@ -6,6 +6,7 @@ import HeaderSearch from './components/header-search.js';
 
 export type HeaderProps = {
   logoLink: string;
+  search_url: string;
```
Comment (pregno) [inline packages/react/ds/src/header/header.tsx:9]: I will set this optional or have another flag to define whether the search is active.
In some cases, the search will not be part of the application, and we will need to hide the button.

**Review by pregno — CHANGES_REQUESTED** [review]

File: packages/react/ds/src/header/header.tsx (hunk 1 above)
Comment (BogdanDanilescu) [reply packages/react/ds/src/header/header.tsx:9]: good idea. I will put this as optional and if the user doesn't use this prop, the search component will not render

**Review by pregno — CHANGES_REQUESTED** [review]
I suggest to review the headers props using something like this.
```
{
  "logo": {
    "image": "...", //optional, url of the image, default is embedded svg
    "href": "/" // required, link of the logo
  },
  "tools": {
    "search": {
      "enable": true, // required, if false we hide the button
      "action": "...", // optional, form action of the search page
      "label": "Search", // optional, default hidded
      "icon": "icon-magnifier", // optional, default icon-magnifier
      "noJsHref": "..." // fallback link for when js is not available
    },
    "menu": {
      "label": "Menu", // optional, default hidded
      "icon": "icon-hamburger", // optional, default icon-hamburger
      "noJsHref": "..." // fallback link for when js is not available
    },
    "items": [ // optional
      { 
        "label": "Home", // optional
        "icon": "icon-home", // required
        "href": "/" // required
      },
      {

      }
    ]
  },
  "links": [
    {
      "label": "nav 1",
      "href": "..."
    },
    {
      "label": "nav 2",
      "href": "..."
    }
  ],
  "languages": [
    {
      "label": "English",
      "href": "..."
    },
    {
      "label": "Gaelic",
      "href": "..."
    }
  ]
}
```

File: packages/react/ds/src/header/header.tsx [hunk 2]
```diff
@@ -6,6 +6,7 @@ import HeaderSearch from './components/header-search.js';
 
 export type HeaderProps = {
   logoLink: string;
+  search_url?: string;
```
Comment (hamza14khan) [inline packages/react/ds/src/header/header.tsx:9]: suggestion: If we can change search_url to camelCase convention?

**Review by hamza14khan — CHANGES_REQUESTED** [review]
Looks good. Just one minor comment and Andrea's feedback.

File: packages/react/ds/src/header/header.tsx [hunk 3]
```diff
@@ -42,13 +66,33 @@ export function Header({ logoLink, languages, navLinks }: HeaderProps) {
...
       >
-        <div className="gi-flex">
-          <a href={logoLink} className="xs:gi-block gi-hidden">
-            <GovieLogo />
+        <div className="gi-flex gi-items-center gi-gap-3">
+          <a href={logo.href} className="xs:gi-block gi-hidden">
+            {logo.image ? (
+              <img
+                className="gi-object-contain gi-h-12"
+                src="https://ds.blocks.gov.ie/_next/static/media/hero.01b46751.png"
```
Comment (pregno) [inline packages/react/ds/src/header/header.tsx:74]: You should add the `{logo.image}` here as `src`

File: packages/react/ds/src/header/header.tsx [hunk 4]
```diff
@@ -42,13 +66,33 @@ export function Header({ logoLink, languages, navLinks }: HeaderProps) {
...
+              <GovieLogo />
+            )}
           </a>
-          <a href={logoLink} className="xs:gi-hidden gi-block">
-            <GovieLogoSmall />
+          <a href={logo.href} className="xs:gi-hidden gi-block">
+            {logo.image ? (
+              <img
+                className="gi-object-contain gi-h-10"
+                src="https://ds.blocks.gov.ie/_next/static/media/hero.01b46751.png"
```
Comment (pregno) [inline packages/react/ds/src/header/header.tsx:84]: Same as above

**Review by pregno — CHANGES_REQUESTED** [review]
LGTM, just a minor fix

File: packages/react/ds/src/header/header.tsx (hunk 3 above)
Comment (BogdanDanilescu) [reply packages/react/ds/src/header/header.tsx]: fixed

**Review by pregno — APPROVED** [review]

**Review by hamza14khan — APPROVED** [review]

**Pull request merged**